web/src/scroll_util.ts

```diff
@@ -1,7 +1,10 @@
 import $ from "jquery";
 import SimpleBar from "simplebar";
 
-export function get_content_element($element) {
+// This type is helpful for testing, where we may have a dummy object instead of an actual jquery object.
+type JQueryOrZJQuery = {__zjquery?: true} & JQuery;
```

Comment:
I was just wondering if it's really necessary to create this type just for testing purposes. Do we actually use types in our tests?

Also, I'm a bit concerned that if we want to add a new test, we'll need to make changes to the actual code and modify the type from JQuery to JQueryOrZJQuery. That seems like it could be a bit of a hassle.

What do you think?

Reply:
Nope, we don't use types in our tests, you could say this was a special case where I needed this to for argument of `get_scroll_element` because we could not mock this function `scroll_util.test.js` when testing for `scroll_element_into_container` but most of the time it will be the case that you will mock this whole module when writing a new test so I don't think it's a problem!

Reply:
Why can't we mock `get_scroll_element` in `scroll_util.test.js`?

Reply:
Because it is now in `scroll_util` itself and we should not mock the module we are testing it will defeat the purpose of tests right? Also, we cannot mutate an ES module from outside we try to override that function manually in test. Hope you understand!

Reply:
By the way, I am open to suggestions if you think we could test the `scroll_element_into_container` function in `scroll_util.test.js` some other better way!

Reply:
I believe we can potentially mock `get_scroll_element` solely for this specific test scenario. My suggestion was inspired by the documentation found [here](https://zulip.readthedocs.io/en/latest/testing/testing-with-node.html#handling-dependencies-in-unit-tests). An example from the documentation illustrates importing the actual code and subsequently defining a new property for a specific function, as shown below:

```javascript
// Import real code.
zrequire('narrow_state');

// And later...
narrow_state.stream = function () {
    return 'office';
};
```

Reply:
This was the very first thing I tried but for some reason when I tried to set a mock function this way it didn't worked it still used the real code.

Reply:
Ah, there's a possibility it may not work, as `scroll_util` relies on the internal usage of `get_scroll_element`.

Reply:
Yeah I'm not sure I see a better way to do this right now.